sentry_sdk/integrations/bottle.py

```diff
@@ -0,0 +1,165 @@
+                            mechanism={"type": "bottle", "handled": self.app.catchall},
+                        )
+                        hub.capture_event(event, hint=hint)
+                        raise exception
+
+                route.call = types.MethodType(patched_call, route)
+                return route, route_args
+
+            self.router.match = types.MethodType(patched_match, self.route)
+
+            # monkey patch method self(Bottle)._handle
+            old_handle = self._handle
+
+            def _patched_handle(self, environ):
+                hub = Hub.current
+                # create new scope
+                scope_manager = hub.push_scope()
+
+                with scope_manager:
+                    app = self
+                    with hub.configure_scope() as scope:
+                        scope._name = "bottle"
+                        scope.add_event_processor(
+                            _make_request_event_processor(
+                                app, bottle_request, integration
+                            )
+                        )
+                    res = old_handle(environ)
+
+                # scope cleanup
+                return res
+
+            self._handle = types.MethodType(_patched_handle, self)
+
+            return SentryWsgiMiddleware(lambda *a, **kw: old_app(self, *a, **kw))(
+                environ, start_response
+            )
+
+        Bottle.__call__ = sentry_patched_wsgi_app  # type: ignore
+
+
+class BottleRequestExtractor(RequestExtractor):
+    def env(self):
+        # type: () -> Dict[str, str]
+        return self.request.environ
+
+    def cookies(self):
+        # type: () -> Dict[str, str]
+        return self.request.cookies
+
+    def raw_data(self):
+        # type: () -> bytes
+        return self.request.body.read()
+
+    def form(self):
+        # type: () -> FormsDict
+        if self.is_json():
+            return None
+        return self.request.forms.decode()
+
+    def files(self):
+        # type: () -> Optional[Dict[str, str]]
+        if self.is_json():
+            return None
+
+        return self.request.files
+
+    def size_of_file(self, file):
+        # type: (FileUpload) -> int
+        return file.content_length
+
+
+def _make_request_event_processor(app, request, integration):
+    # type: (Bottle, LocalRequest, BottleIntegration) -> Callable
+    def inner(event, hint):
+        # type: (Dict[str, Any], Dict[str, Any]) -> Dict[str, Any]
+
+        # if the request is gone we are fine not logging the data from
+        # it.  This might happen if the processor is pushed away to
+        # another thread.
+        if request is None:
+            return event
+
+        try:
+            if integration.transaction_style == "endpoint":
+                event["transaction"] = request.route.name or getattr(
+                    request.route.callback, "func_name", request.route.callback.__name__
```

Comment:
I think the function `transaction_from_function` is useful here. It creates a full path including module etc

Reply:
Thanks for the hint. The function seems quite handy here.